dvc/repo/ls.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+from dvc.exceptions import DvcException, NotDvcRepoError, OutputNotFoundError
+from dvc.scm.base import CloneError
+
+
+@staticmethod
+def ls(url, target=None, rev=None, recursive=None, outputs_only=False):
+    if os.path.exists(url):
+        if rev:
+            raise DvcException("Local repo doesn't support revisions")
+        return _ls_local_dir(url, target, recursive, outputs_only)
+
+    try:
+        return _ls_remote_git_repo(url, target, rev, recursive, outputs_only)
+    except CloneError:
+        raise DvcException("There is no such url {}".format(url))
```

Comment:
CloneError doesn't mean that there is no such url, it only means that we were not able to clone it (e.g. because of access issues). Even if we decide to wrap it, it is not suitable to do it here, it should be in `external_repo` or somewhere there.

Reply:
It's an attempt to raise an error on non existed repo https://github.com/iterative/dvc/pull/3246#discussion_r375929735 but you are right. One problem that needs to extend errors for getting more info. The reason I've caught and raise the error here cause I would like to point that `url` that was requested in `dvc list url` doesn't exist. So I think it's right place for it, but I need to get more info from `dvc.scm`

Reply:
Not sure I follow how that relates to that comment. In that comment I was talking about path within the repo that doesn't exist, not the url of the repo that doesn't exist.